test/e2e/storage/volume_expand.go

```diff
@@ -0,0 +1,196 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = SIGDescribe("Volume expand [Slow]", func() {
+	var (
+		c           clientset.Interface
+		ns          string
+		err         error
+		pvc         *v1.PersistentVolumeClaim
+		resizableSc *storage.StorageClass
+	)
+
+	f := framework.NewDefaultFramework("volume-expand")
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("aws", "gce")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+		resizableSc, err = createResizableStorageClass(test, ns, "resizing", c)
+		Expect(err).NotTo(HaveOccurred(), "Error creating resizable storage class")
+		Expect(*resizableSc.AllowVolumeExpansion).To(BeTrue())
+
+		pvc = newClaim(test, ns, "default")
+		pvc.Spec.StorageClassName = &resizableSc.Name
+		pvc, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		Expect(err).NotTo(HaveOccurred(), "Error creating pvc")
+	})
+
+	AfterEach(func() {
+		framework.DeletePersistentVolumeClaim(c, pvc.Name, pvc.Namespace)
+		framework.ExpectNoError(c.StorageV1().StorageClasses().Delete(resizableSc.Name, nil))
+	})
+
+	It("Verify if editing PVC allows resize", func() {
+		By("Waiting for pvc to be in bound phase")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
+		Expect(len(pvs)).To(Equal(1))
+
+		By("Creating a pod with volume")
+		pod, err := framework.CreatePod(c, ns, nil, pvcClaims, false, "")
+		Expect(err).NotTo(HaveOccurred(), "While creating pods for resizing")
+
+		By("Expanding current pvc")
+		newSize := resource.MustParse("6Gi")
+		pvc, err = expandPVCSize(pvc, newSize, c)
+		Expect(err).NotTo(HaveOccurred(), "While updating pvc for more size")
+		Expect(pvc).NotTo(BeNil())
+
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("Error updating pvc size %s", pvc.Name)
+		}
+
+		By("Waiting for cloudprovider resize to finish")
+		err = waitForControllerVolumeResize(pvc, c)
```

Comment:
Maybe this func should return both *pvc and err, like `waitForFSResize`does below? This would save you from re-getting the pvc next.

Reply:
We don't poll for PVC in this function. We only poll for PV and that is why this function can't return updated pvc. We can of course update the function to poll for PVC as well, but it will be waste of API resources because this function does not care about pvc state.